clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -522,7 +522,12 @@ private long waitOnMetadata(String topic, long maxWaitMs) throws InterruptedExce
             log.trace("Requesting metadata update for topic {}.", topic);
             int version = metadata.requestUpdate();
             sender.wakeup();
-            metadata.awaitUpdate(version, remainingWaitMs);
+            try {
+                metadata.awaitUpdate(version, remainingWaitMs);
```

Comment:
Is it actually useful for `metadata.awaitUpdate` to throw an exception? Maybe it should simply return a `boolean`.